### PR TITLE
feat: continue expanding use of tiered badge data

### PR DESCRIPTION
**Pull request by dyersituations**
https://kiva.atlassian.net/browse/MP-966

- Next step is getting the badges section of MyKiva to show the correct badge title, level, and icon based on the current in-progress tier
- Added unit testing for new composable (mostly adding lots of mock data to make a realistic test)
- Fixed small spacing issue on MyKiva badge section badges
- Moved tier date cleanup to composable
- Moved Vue lifecycle methods out of composable so it could be reused as a util
- Other minor cleanup, like commenting

image

### CODE REVIEW

File: src/composables/useBadgeData.js [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
 	});
 
-	const fetchAchievementData = () => {
+	/**
+	 * Calls Apollo to get the badge achievement service data
+	 */
+	const fetchAchievementData = apollo => {
 		apollo.query({ query: userAchievementProgressQuery })
 			.then(result => {
 				badgeAchievementData.value = result.data?.userAchievementProgress?.tieredLendingAchievements ?? [];
```
Comment (roger-in-kiva) [inline src/composables/useBadgeData.js:35]: Can we add `lendingAchievements` here?

Comment (dyersituations) [reply src/composables/useBadgeData.js:35]: Wouldn't we need to use both to get all achievements (old and new)?

Comment (roger-in-kiva) [reply src/composables/useBadgeData.js:35]: We need both, for the journey and earned badges, these last ones includes both.
Probably we can also just call the new field `achievements` on the query, to get both in a call.

Comment (dyersituations) [reply src/composables/useBadgeData.js:35]: What do you mean by the new field `achievements`? I'm not seeing that in the gql sandbox.

Comment (roger-in-kiva) [reply src/composables/useBadgeData.js:35]: For real? That one includes both cases.

Screenshot 2024-10-28 at 5 08 25 p m

Comment (dyersituations) [reply src/composables/useBadgeData.js:35]: Ah, on the parent, got it, thanks.

I'll do this in a follow-up PR.

Comment (roger-in-kiva) [reply src/composables/useBadgeData.js:35]: Thanks!

**Review by roger-in-kiva — APPROVED** [review]

**Pull request merged**